cluster/pre-existing/util.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library of helper functions for a pre-existing Kubernetes cluster
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${KUBE_ROOT}/cluster/common.sh"
+source "${KUBE_ROOT}/cluster/lib/util.sh"
+
+function detect-project() {
+    if [[ -z "${MASTER_IP-}" ]]; then
+	echo "Set 'MASTER_IP' to the instance assigned to be the Kubernetes master" 1>&2
+	exit 1
+    fi
+
+    if [[ -z "${PROJECT-}" ]]; then
+	PROJECT="kubemark"
+    fi
+
+    if [[ -z "${SERVICE_CLUSTER_IP_RANGE-}" ]]; then
+	octets=($(echo "${MASTER_IP}" | sed -e 's|/.*||' -e 's/\./ /g'))
+	octets[3]=0
+	cluster_range=$(echo "${octets[*]}" | sed 's/ /./g')
+	SERVICE_CLUSTER_IP_RANGE="${SERVICE_CLUSTER_IP_RANGE:-$cluster_range}"
+    fi
+}
+
+function run-cmd-with-retries {
```

Comment:
Some bugs around this function's logic have been fixed recently. Please incorporate them.